compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5382,12 +5382,15 @@ private void checkArrayLibSortFuncArgs(BLangInvocation iExpr) {
             returnType = ((BLangSimpleVarRef) keyFunction).type.getReturnType();
         } else if (keyFunction.getKind() == NodeKind.ARROW_EXPR) {
             BLangArrowFunction arrowFunction = ((BLangArrowFunction) keyFunction);
-            pos = arrowFunction.params.get(0).pos;
-            returnType = arrowFunction.params.get(0).type;
+            pos = arrowFunction.body.expr.pos;
+            returnType = arrowFunction.body.expr.type;
         } else {
             BLangLambdaFunction keyLambdaFunction = (BLangLambdaFunction) keyFunction;
             pos = keyLambdaFunction.function.pos;
             returnType = keyLambdaFunction.function.type.getReturnType();
+            if (returnType.tag == TypeTags.SEMANTIC_ERROR) {
```

Comment:
Do we have tests covering this? Is this only an issue with lambda functions?

Reply:
Added a test. It's an issue with arrow functions